SC_redesign/Crop_and_Soil/soil/evaporation.py

```diff
@@ -0,0 +1,220 @@
+from math import exp
+from typing import Optional
+from SC_redesign.Crop_and_Soil.soil.soil_data import SoilData
+
+"""
+This module is based off of the 'Soil Water Evaporation' (2:2.3.3.2) section of the SWAT Theoretical documentation.
+"""
+
+
+class Evaporation:
+    def __init__(self, soil_data: Optional[SoilData], field_size: Optional[float] = None):
+        """This method initializes the SoilData object that this module will work with, or create one if none provided.
+
+        Parameters
+        ----------
+        soil_data : SoilData, optional
+            The SoilData object used by this module to track evaporation and transpiration from the soil profile,
+            creates new one if one is not provided.
+        field_size : float, optional
+            Used to initialize a SoilData object for this module to work with, if a pre-configured SoilData object is
+            not provided (ha)
+
+        """
+        self.data = soil_data or SoilData(field_size=field_size)
+
+    def evaporate(self, maximum_soil_water_evaporation: float) -> None:
+        """Evaporates water from the soil profile.
+
+        Parameters
+        ----------
+        maximum_soil_water_evaporation : float
+            Maximum amount of water allowed to be evaporated from the soil profile on the current day (mm)
+
+        Notes
+        -----
+        This method takes in a maximum amount of water that may be evaporated from the soil profile, then iterates over
+        the layers in the soil profile and determines how much to evaporate from that layer. If the amount determined to
+        be evaporated would put the total amount of evaporation over the limit, the amount is reduced to meet the limit
+        exactly and the loop is exited after evaporating the reduced amount from the current layer.
+
+        """
+        amount_available_for_evaporation = maximum_soil_water_evaporation
+        for layer in self.data.soil_layers:
+            evaporative_demand = self._determine_layer_evaporative_demand(
+                maximum_soil_water_evaporation, layer.top_depth, layer.bottom_depth,
+                layer.soil_evaporation_compensation_coefficient)
+            evaporative_demand_reduced = self._determine_evaporative_demand_reduced(
+                evaporative_demand, layer.water_content, layer.field_capacity_content, layer.wilting_point_content)
+            amount_water_removed = self._determine_amount_water_removed(
+                evaporative_demand_reduced, layer.water_content, layer.wilting_point_content)
+
+            maximum_amount_evaporated = amount_available_for_evaporation <= amount_water_removed
+            if maximum_amount_evaporated:
+                amount_water_removed = amount_available_for_evaporation
+            layer.water_content -= amount_water_removed
+            amount_available_for_evaporation -= amount_water_removed
+            if maximum_amount_evaporated:
+                break
```

Comment:
```suggestion
      if maximum_amount_evaporated:
            amount_water_removed = amount_available_for_evaporation
            layer.water_content -= amount_water_removed
            amount_available_for_evaporation -= amount_water_removed
            break
      else:
            layer.water_content -= amount_water_removed
            amount_available_for_evaporation -= amount_water_removed
```
Although there are more lines, I think that reducing condition statements improves readability. I don't think it's a big deal so up to you tho.

Reply:
This is a good suggestion. I did not go with the duplicate lines but I found a way to accomplish the same thing with a `min` statement and a single `if` statement.